exonum/src/encoding/protobuf/mod.rs

```diff
@@ -78,7 +78,11 @@ impl ProtobufConvert for crypto::Hash {
 
     fn from_pb(pb: Hash) -> Result<Self, ()> {
         let data = pb.get_data();
-        crypto::Hash::from_slice(data).ok_or(())
+        if data.len() == crypto::HASH_SIZE {
```

Comment:
Why do we need these checks? [`Hash` (and other types from sodiumoxide) seem to do their own length checks](https://github.com/sodiumoxide/sodiumoxide/blob/bdc268f59e7465cbff85cd5e58f075f75504ffab/src/newtype_macros.rs#L17-L19).